src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -101,13 +101,17 @@ public Checksum CreateChecksum(AnalyzerReference reference, CancellationToken ca
                         WriteUnresolvedAnalyzerReferenceTo(unresolved, writer);
                         break;
 
+                    case AnalyzerReference analyzerReference when analyzerReference.GetType().FullName == "Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioUnresolvedAnalyzerReference":
+                        WriteUnresolvedAnalyzerReferenceTo(analyzerReference, writer);
+                        break;
```

Comment:
I really don't like this, but not sure what else to do, since `VisualStudioUnresolvedAnalyzerReference` is defined at a higher layer.

@CyrusNajmabadi suggested just not serializing it at all instead of throwing.  @heejaechang - thoughts?